.github/workflows/Release-Please.yml

```diff
@@ -0,0 +1,67 @@
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: write
+  pull-requests: write
+
+name: Run Release Please
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    environment: deployment
+    strategy:
+      matrix:
+        node-version: [ 18.x ]
+    steps:
+      - uses: google-github-actions/release-please-action@v2
+        id: release
+        with:
+          command: manifest
+          token: ${{secrets.GITHUB_TOKEN}}
+          default-branch: main
+          monorepo-tags: true
+
+
+  Setup-Environment:
+    runs-on: ubuntu-latest
+    if: ${{ steps.release.outputs.releases_created }}
+    strategy:
+      matrix:
+        node-version: [ 18.x ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+  Build:
+    runs-on: ubuntu-latest
+    if: ${{ steps.release.outputs.releases_created }}
+    needs: [ Setup-Environment ]
+    steps:
+      - run: yarn build
+
+  npm-publish:
```

Comment:
How does this work? Configured in this way, is npm-publish dependent on Build? Or should build be a step in Publish? I guess the same would be true for the release-please job, so maybe I'm misunderstanding actions..

Reply:
The 'needs' bit of the configuration means that it will wait until that job has completed